File: subroutines/all/subroutine project.f
      subroutine project(x,c,ax,p,a)
      dimension x(3),c(3),ax(3),p(3)
c     Find the nearest point p from x on c+a*ax
      a=scprod(ax,x)-scprod(ax,c)
      do k=1,3
        p(k)=c(k)+a*ax(k)
      end do
      return
      end

File: fortran to python conversion/dependencies/subroutine project.py
import numpy as np

def project(x,c,ax):
	"""
	
	Find the nearest point p from x on c+a*ax
	
	***
	Parameters:
	x, c, ax: three vectors

	Returns:
	p: the nearest point 
	a : ax|(x-c)

	"""
	a = np.dot(ax,x-c)
	for k in range(3):
		p[k] = c[k] + a*ax[k]
	return p,a 